nautobot/dcim/views.py

```diff
@@ -269,62 +358,102 @@ class LocationUIViewSet(NautobotUIViewSet):
     bulk_update_form_class = forms.LocationBulkEditForm
     serializer_class = serializers.LocationSerializer
 
+    object_detail_content = object_detail.ObjectDetailContent(
+        panels=(
+            object_detail.ObjectFieldsPanel(
+                weight=100,
+                section=SectionChoices.LEFT_HALF,
+                fields=[
+                    "location_type",
+                    "status",
+                    "tenant",
+                    "facility",
+                    "asn",
+                    "time_zone",
+                    "description",
+                ],
+                value_transforms={
```

Comment:
We need to add heirarchy (via ancestors) and then likely update `render_key` on a custom `ObjectFieldsPanel`. 

Then transform, based on this logic"


```html
        {% for ancestor in ancestors %}
            <li>{{ ancestor|hyperlinked_object:"name" }} ({{ ancestor.location_type|hyperlinked_object:"name" }})
                <ul>
        {% endfor %}
        <li><strong>{{ location|hyperlinked_object:"name" }} ({{ location.location_type|hyperlinked_object:"name" }})</strong></li>
        {% for ancestor in ancestors %}
                </ul>
            </li>

```